tools/kagami/src/genesis.rs

```diff
@@ -241,10 +259,231 @@ fn generate_synthetic(
+                .wrap_err_with(|| {
+                    eyre!(
+                        "Failed to deserialize public key from multihash string for algorithm `{}`",
+                        self.algorithm
+                    )
+                }),
+        }
+    }
+}
+
+impl<T: Write> RunArgs<T> for SignArgs {
+    fn run(self, writer: &mut BufWriter<T>) -> Outcome {
+        let key_pair = if let Some(path) = self.keypair_file {
+            read_keypair(path)?
+        } else {
+            let (public_key, private_key) = (self.get_public_key()?, self.get_private_key()?);
+            KeyPair::new(public_key, private_key)?
+        };
+
+        let genesis_block = RawGenesisBlock::from_path(&self.genesis_file)?;
+        let genesis_signature =
+            GenesisNetwork::new_genesis_signature(genesis_block, &self.chain_id, &key_pair);
+        let genesis_signature_string = genesis_signature.to_hex_string();
+
+        let scale_encoded_signature = self.binary.then(|| genesis_signature_string.encode());
+
+        if let Some(path) = self.out_file {
+            if let Some(encoded_genesis_signature) = scale_encoded_signature {
+                fs::write(&path, encoded_genesis_signature)?;
+            } else {
+                fs::write(&path, genesis_signature_string)?;
+            }
+
+            writeln!(
+                writer,
+                "Genesis was successfully signed and written to `{}`",
+                path.display()
+            )?;
+        } else if self.binary {
+            writeln!(
+                writer,
+                "SCALE encoded data is not supported for console outputs."
+            )?;
+        } else {
+            writeln!(writer, "{genesis_signature_string}")?;
+        }
+
+        Ok(())
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use iroha_genesis::{GenesisSignature, GenesisTransaction};
+
+    use super::*;
+    use crate::crypto::AlgorithmArg;
+
+    const GENESIS_JSON_PATH: &str = "../../configs/swarm/genesis.json";
+
+    #[test]
+    fn genesis_signing_works() -> Result<()> {
+        let keypair_config = crypto::Args {
+            algorithm: AlgorithmArg::default(),
+            private_key: None,
+            seed: None,
+            json: true,
+            compact: false,
+        };
+
+        let mut keypair_json = BufWriter::new(Vec::new());
+        keypair_config.run(&mut keypair_json)?;
+        let keypair: KeyPair = serde_json::from_slice(keypair_json.buffer())?;
+
+        let tmp_keypair_json_file = tempfile::NamedTempFile::new()?;
+        fs::write(tmp_keypair_json_file.path(), keypair_json.buffer())?;
+
+        let chain_id_slice = "0123456";
+        let chain_id = ChainId::from(chain_id_slice);
+        let crypto_genesis_config = SignArgs::try_parse_from([
+            "sign",
+            "--chain-id",
+            chain_id_slice,
+            "--genesis-file",
+            GENESIS_JSON_PATH,
```

Comment:
```suggestion
            GENESIS_JSON_PATH,
```

why isn't this positional argument?

Reply:
I don't have an answer to that. It seems like we don't positional args in both swarm and kagami and I thought not to break the "tradition".
I thought specifying the argument explicitly makes it less error prone, don't you agree?
@0x009922 what do you think?

Reply:
I think either way is fine, but I have a _feeling_ that explicit argument would be more consistent overall.

Reply:
I have a _feeling_ that if you're signing something, it's not optional and should be a positional arg.
CLI apps should have one positional argument, that is the object of operation